### PR TITLE
fix gracefilly shutdown bug, issue #958

### PR DESCRIPTION
**Pull request by alpha-baby**
### unit test

```bash
$ go test . -v -run=^TestShutdown$
=== RUN   TestShutdown
=== PAUSE TestShutdown
=== CONT  TestShutdown
--- PASS: TestShutdown (0.51s)
PASS
ok      github.com/valyala/fasthttp     0.530s
```

### CODE REVIEW

**Comment by erikdubbelboer** [comment]
See: https://github.com/valyala/fasthttp/issues/958#issuecomment-774428181

File: server.go [hunk 1]
```diff
@@ -375,11 +375,17 @@ type Server struct {
 	// which will close it when needed.
 	KeepHijackedConns bool
 
+
+	// CloseOnShutdown , when set true server add HTTP Header: `Connection: close` into HTTP Response
+	// refer isuee https://github.com/valyala/fasthttp/issues/958
```
Comment (erikdubbelboer) [inline server.go:380]:
```suggestion
	// CloseOnShutdown when true adds a `Connection: close` header when when the server is shutting down.
```

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

**Comment by alpha-baby** [comment]
@erikdubbelboer can you fix this unit test, becase i'm not hava windwos PC.

**Comment by erikdubbelboer** [comment]
The tests are a bit flaky, everything seems fine so I'm merging this. Thanks!

**Pull request merged**